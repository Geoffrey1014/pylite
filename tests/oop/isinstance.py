class Animal:
    pass

class Dog(Animal):
    def __init__(self, name):
        self.name = name

    def bark(self):
        print "%s: bark" % self.name

class Cat(Animal):
    def __init__(self, name):
        self.name = name

    def meow(self):
        print "%s: meow" % self.name

puppy = Dog("Puppy")
tom = Cat("Tom")

print isinstance(puppy, Animal)
print isinstance(puppy, Dog)
print isinstance(puppy, Cat)

print isinstance(tom, Animal)
print isinstance(tom, Cat)
print isinstance(tom, Dog)

animals = [puppy, "String", tom]
for animal in animals:
    if isinstance(animal, Dog):
        animal.bark()
    if isinstance(animal, Cat):
        animal.meow()
