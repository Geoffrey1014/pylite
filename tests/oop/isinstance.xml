lambda
  name __init__
  param_count 0
  local_count 1
  label_count 7
  insn_vector
    new_class c0
    store_global c0
    new_class c1
    store_global c1
    load_global c0
    load_global c1
    inherit
    load_const c2
    load_global c1
    store_field c3
    load_const c4
    load_global c1
    store_field c5
    new_class c6
    store_global c6
    load_global c0
    load_global c6
    inherit
    load_const c7
    load_global c6
    store_field c3
    load_const c8
    load_global c6
    store_field c9
    load_const c10
    load_global c1
    call 1
    store_global c11
    load_const c12
    load_global c6
    call 1
    store_global c13
    load_global c0
    load_global c11
    load_global c14
    call 2
    print
    load_global c1
    load_global c11
    load_global c14
    call 2
    print
    load_global c6
    load_global c11
    load_global c14
    call 2
    print
    load_global c0
    load_global c13
    load_global c14
    call 2
    print
    load_global c6
    load_global c13
    load_global c14
    call 2
    print
    load_global c1
    load_global c13
    load_global c14
    call 2
    print
    new_list 3
    load_global c11
    load_const c15
    dup_top 2
    store_item
    load_const c16
    load_const c17
    dup_top 2
    store_item
    load_global c13
    load_const c18
    dup_top 2
    store_item
    store_global c19
    load_global c19
    iterator
    store_local -1
    jmp L1
    label L0
    store_global c20
    load_global c1
    load_global c20
    load_global c14
    call 2
    jfalse L4
    load_global c20
    load_field c5
    call 0
    pop
    jmp L3
    label L4
    label L3
    load_global c6
    load_global c20
    load_global c14
    call 2
    jfalse L6
    load_global c20
    load_field c9
    call 0
    pop
    jmp L5
    label L6
    label L5
    label L1
    load_local -1
    next L0
    pop
    label L2
    load_const c21
    return
  const_vector
    Animal
    Dog
    lambda
      name __init__
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        store_field c0
        load_const c1
        return
      const_vector
        name
        None
      guard_vector
      line_number_map 6 6 6 5 5
    __init__
    lambda
      name bark
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_field c0
        load_const c1
        mod
        print
        load_const c2
        return
      const_vector
        name
        "%s: bark"
        None
      guard_vector
      line_number_map 9 9 9 9 9 8 8
    bark
    Cat
    lambda
      name __init__
      param_count 2
      local_count 0
      label_count 0
      insn_vector
        load_local 1
        load_local 0
        store_field c0
        load_const c1
        return
      const_vector
        name
        None
      guard_vector
      line_number_map 13 13 13 12 12
    lambda
      name meow
      param_count 1
      local_count 0
      label_count 0
      insn_vector
        load_local 0
        load_field c0
        load_const c1
        mod
        print
        load_const c2
        return
      const_vector
        name
        "%s: meow"
        None
      guard_vector
      line_number_map 16 16 16 16 16 15 15
    meow
    "Puppy"
    puppy
    "Tom"
    tom
    isinstance
    0.000000
    "String"
    1.000000
    2.000000
    animals
    animal
    None
  guard_vector
  line_number_map 1 1 4 4 4 4 4 5 5 5 8 8 8 11 11 11 11 11 12 12 12 15 15 15 18 18 18 18 19 19 19 19 21 21 21 21 21 22 22 22 22 22 23 23 23 23 23 25 25 25 25 25 26 26 26 26 26 27 27 27 27 27 29 29 29 29 29 29 29 29 29 29 29 29 29 29 30 30 30 30 30 30 31 31 31 31 31 32 32 32 32 31 31 31 33 33 33 33 33 34 34 34 34 33 33 33 30 30 30 30 30 1 1
