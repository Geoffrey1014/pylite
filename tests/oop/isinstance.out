True
True
False
True
True
False
Puppy: bark
Tom: meow
